bundles/org.openhab.binding.gree/src/main/java/org/openhab/binding/gree/internal/handler/GreeAirDevice.java

```diff
@@ -0,0 +1,530 @@
+
+            // Encrypt and send the Status Request pack
+            String encryptedStatusReqPacket = GreeCryptoUtil.encryptPack(getKey().getBytes(), reqStatusPackStr);
+            DatagramPacket sendPacket = createPackRequest(0,
+                    new String(encryptedStatusReqPacket.getBytes(), StandardCharsets.UTF_8));
+            clientSocket.send(sendPacket);
+
+            // Recieve a response
+            JsonReader receivedData = receiveResponse(clientSocket);
+
+            // Keep a copy of the old response to be used to check if values have changed
+            // If first time running, there will not be a previous GreeStatusResponsePack4Gson
+            if (statusResponseGson.isPresent() && statusResponseGson.get().packJson != null) {
+                prevStatusResponsePackGson = Optional
+                        .of(new GreeStatusResponsePackDTO(statusResponseGson.get().packJson));
+            }
+
+            // Read the response, create the JSON to hold the response values
+            GreeStatusResponseDTO resp = gson.fromJson(receivedData, GreeStatusResponseDTO.class);
+            resp.decryptedPack = GreeCryptoUtil.decryptPack(this.getKey().getBytes(), resp.pack);
+            logger.debug("Response from device: {}", resp.decryptedPack);
+            resp.packJson = gson.fromJson(new JsonReader(new StringReader(resp.decryptedPack)),
+                    GreeStatusResponsePackDTO.class);
+
+            // save the results
+            statusResponseGson = Optional.of(resp);
+            updateTempFtoC();
+        } catch (IOException e) {
+            throw new GreeException("I/O exception while receiving data", e);
+        } catch (RuntimeException e) {
+            String json = statusResponseGson.map(r -> r.packJson.toString()).orElse("n/a");
+            throw new GreeException("Exception while receiving data, JSON=" + json, e);
+        }
+    }
+
+    public void bindWithDevice(DatagramSocket clientSocket) throws GreeException {
+        try {
+            // Prep the Binding Request pack
+            GreeBindRequestPackDTO bindReqPackGson = new GreeBindRequestPackDTO();
+            bindReqPackGson.mac = getId();
+            bindReqPackGson.t = GREE_CMDT_BIND;
+            bindReqPackGson.uid = 0;
+            String bindReqPackStr = gson.toJson(bindReqPackGson);
+
+            // Encrypt and send the Binding Request pack
+            String encryptedBindReqPacket = GreeCryptoUtil.encryptPack(GreeCryptoUtil.getAESGeneralKeyByteArray(),
+                    bindReqPackStr);
+            DatagramPacket sendPacket = createPackRequest(1,
+                    new String(encryptedBindReqPacket.getBytes(), StandardCharsets.UTF_8));
+            clientSocket.send(sendPacket);
+
+            // Recieve a response, create the JSON to hold the response values
+            GreeBindResponseDTO resp = gson.fromJson(receiveResponse(clientSocket), GreeBindResponseDTO.class);
```

Comment:
You should catch the json syntax exception

Reply:
done